sympy/functions/combinatorial/tests/test_comb_factorials.py

```diff
@@ -548,6 +548,54 @@ def test_binomial_rewrite():
     assert binomial(n, k).rewrite(ff) == ff(n, k) / factorial(k)
     assert binomial(n, x).rewrite(ff) == binomial(n, x)
 
+def test_multinomial():
+    x = Symbol('x')
+    n = Symbol('n', integer=True)
+    nz = Symbol('nz', integer=True, nonzero=True)
+    k = Symbol('k', integer=True)
+    kp = Symbol('kp', integer=True, positive=True)
+    kn = Symbol('kn', integer=True, negative=True)
+    u = Symbol('u', negative=True)
+    v = Symbol('v', nonnegative=True)
+    p = Symbol('p', positive=True)
+    z = Symbol('z', zero=True)
+    nt = Symbol('nt', integer=False)
+    kt = Symbol('kt', integer=False)
+    a = Symbol('a', integer=True, nonnegative=True)
+    b = Symbol('b', integer=True, nonnegative=True)
+
+    assert multinomial(0, 0) == 1
+    assert multinomial(1, 1) == 2
+    assert multinomial(1, 1, 1) == 6
+    assert multinomial(n, z) == gamma(n + z + 1)/(gamma(n + 1)*gamma(z + 1))
+    assert multinomial(1, 2) == 3
+    assert multinomial(-1, 2) == 0
+    assert multinomial(1, -1) == 0
+    assert multinomial(S.Half, S.Half) == 4/pi
+    assert multinomial(n, -1) == 0 # holds for all integers (negative, zero, positive)
+    assert multinomial(kp, -1) == 0
+    assert multinomial(nz, 0) == 1
+    assert expand_func(multinomial(n, 1)) == n + 2
+    assert expand_func(multinomial(n, 2)) == (n + 1)*(n + 2)/2
+    assert expand_func(multinomial(n, 2, 3)) == (n + 1)*(n + 2)*(n + 3)*(n + 4)*(n + 5)/12
+    assert multinomial(n, 3).expand(func=True) ==  n**3/6 + n**2 + 11*n/6 + 1
+    assert expand_func(multinomial(n, 3)) ==  (n + 1)*(n + 2)*(n + 3)/6
+    assert multinomial(kp, kp + 1) == gamma(2*kp + 2)/(gamma(kp + 1)*gamma(kp + 2))
+    assert expand_func(multinomial(n, n - 3)) == (n - 2)*(n - 1)*gamma(2*n)/(n*(2*n - 2)*(2*n - 1)*gamma(n)**2)
+    assert multinomial(nt, k).is_integer is None
+    assert multinomial(gamma(25), 6) == 3171544675936039897227550093537
+    assert multinomial(1324, 47) == 4805336849842017230605644547972993864468963766492826697060943517257216750255923254377400
+    assert multinomial(1735, 43) == 554080992355860878540612726971346089180030283930872963382734453926471927466461350797200
+    assert multinomial(2512, 53) == 654306322457634362990948426584840437888409225427582147525467248791790713032071278685309353241280812107316446880
+    assert multinomial(3383, 52) == 62093899686661978480461599007894723856068140836407630133468085365409808296508050424855664272291741940016742401162520
+    assert multinomial(4321, 51) == 227457163650726634405257688425597243324930864330979686419067341046916573628729411790197413544787445180629754503984973472
```

Comment:
Some simple tests should do. I'd suggest to use small numbers so that it's easy to maintain the code in future.

Reply:
I saw that binomials had very large numbers in it's test cases ... wouldn't it be required to check that the code doesn't break down while using large numbers ?
or should I remove them ?

Reply:
I see, IMO, it should be okay to have one such test case as a corner case but having too many makes it difficult to verify the outputs.

Reply:
Awesome ... will reduce the number of cases for big numbers then